pkg/apis/core/types_cloudprofile.go

```diff
@@ -147,10 +149,20 @@ type ExpirableVersion struct {
 	// Version is the version identifier.
 	Version string
 	// ExpirationDate defines the time at which this version expires.
+	// Deprecated: Is replaced by Lifecycle.
 	ExpirationDate *metav1.Time
 	// Classification defines the state of a version (preview, supported, deprecated).
-	// To get the currently valid classification, use CurrentLifecycleClassification().
+	// Deprecated: Is replaced by Lifecycle.
 	Classification *VersionClassification
+	// Lifecycle defines the lifecycle stages for this version. Cannot be used in combination with deprecated Classification and ExpirationDate.
+	Lifecycle []LifecycleStage
+}
+
+type LifecycleStage struct {
```

Comment:
Requires docstring. See `v1beta1/types_cloudprofile.go`:
```suggestion
// LifecycleStage describes a stage in the classification lifecycle.
type LifecycleStage struct {
```